src/utils/GraphQLStream.js

```diff
@@ -0,0 +1,184 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { lex, Source, getLocation } from 'graphql/language';
+import { getTokenKindDesc } from 'graphql/language/lexer';
+
+/**
+ * GraphQLStream implements a stream of string tokens given a GraphQL source
+ * text, designed to work with GraphQL reference implementations
+ * (https://github.com/graphql/graphql-js). The API design follows that of
+ * CodeMirror.StringStream.
+ *
+ * Required:
+ *
+ *      sourceText: (string), A raw GraphQL source text. Works best if a line
+ *        is supplied.
+ *
+ */
+
+type GraphQLToken = {
+  kind: number,
+  start: number,
+  end: number,
+  value?: string,
+};
+
+export default class GraphQLStream {
+  constructor(sourceText: string): void {
+    // _lastTokenStartPosition preserves the current position of the lexer
+    // in case the stream position needs to be manipulated.
+    this._lastTokenStartPosition = 0;
+    this._sourceText = sourceText;
+    this._lex = lex(new Source(sourceText));
```

Comment:
I don't think this is necessary - this shouldn't require GraphQL's lexer - the "onlineParser" is responsible for lexing.

This is actually important because onlineParser should be able to do partial lexing and `graphql`'s lexer is not designed for that. I'd suggest taking a closer look at what codemirror is doing here - it's likely simpler